### PR TITLE
kv/client: cherry pick #1591,#1674,#1682,#1481,#1847,#1903 to release-5.0

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

cherry pick #1591,#1674,#1682,#1481,#1847,#1903 to release-5.0

### What is changed and how it works?

cherry pick #1591,#1674,#1682,#1481,#1847,#1903 to release-5.0

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- No release note

### CODE REVIEW

File: cdc/kv/client.go [hunk 1]
```diff
@@ -68,15 +68,15 @@ const (
...
 )
 
+// time interval to force kv client to terminate gRPC stream and reconnect
+var reconnectInterval = 15 * time.Minute
+
 // hard code switch
 // true: use kv client v2, which has a region worker for each stream
 // false: use kv client v1, which runs a goroutine for every single region
-var enableKVClientV2 = false
+var enableKVClientV2 = true
```
Comment (amyangfei) [inline cdc/kv/client.go:79]: enable it for test

**Review by overvenus — APPROVED** [review]
LGTM

**Review by leoppro — APPROVED** [review]

**Pull request merged**